File: src/main/java/com/jakewharton/byteunits/UnitUtil.java
package com.jakewharton.byteunits;

import java.text.DecimalFormat;
import java.text.NumberFormat;

final class UnitUtil {
  static final NumberFormat DEFAULT_FORMAT = new DecimalFormat("#,##0.#");

  /** Multiply {@code size} by {@code factor} accounting for overflow. */
  static long multiply(long size, long factor, long over) {
    if (size > over) {
      return Long.MAX_VALUE;
    }
    if (size < -over) {
      return Long.MIN_VALUE;
    }
    return size * factor;
  }

  private UnitUtil() {
    throw new AssertionError("No instances.");
  }
}

File: src/main/java/com/jakewharton/byteunits/UnitUtil.kt
package com.jakewharton.byteunits

internal const val DEFAULT_FORMAT_PATTERN = "#,##0.#"

/** Multiply [size] by [factor] accounting for overflow. */
internal fun multiply(size: Long, factor: Long, over: Long): Long = when {
  size > over -> Long.MAX_VALUE
  size < -over -> Long.MIN_VALUE
  else -> size * factor
}
